library/std/src/process/tests.rs

```diff
@@ -378,6 +378,42 @@ fn test_interior_nul_in_env_value_is_error() {
     }
 }
 
+#[test]
+fn test_empty_env_key_is_error() {
+    match env_cmd().env("", "value").spawn() {
+        Err(e) => assert_eq!(e.kind(), ErrorKind::InvalidInput),
+        Ok(_) => panic!(),
+    }
+}
+
+#[test]
+fn test_interior_equals_in_env_key_is_error() {
+    match env_cmd().env("has=equals", "value").spawn() {
+        Err(e) => assert_eq!(e.kind(), ErrorKind::InvalidInput),
+        Ok(_) => panic!(),
+    }
+}
```

Comment:
Weird, https://learn.microsoft.com/en-us/windows/win32/procthread/environment-variables `The name of an environment variable cannot include an equal sign (=).`

Can this have test:

```rust
#[test]
fn test_env_name_starts_with_equals_is_error() {
    match env_cmd().env("=startsequals", "value").spawn() {
        Err(e) => assert_eq!(e.kind(), ErrorKind::InvalidInput),
        Ok(_) => panic!(),
    }
}
```

Reply:
Shouldn't test for `env_cmd().env("has=equals", "value")` check what value actually set? I expect that this will be `has` with value `=equalsvalue`

Reply:
I don't understand what you're asking me to do.

There's a test for interior equals sign (above) that checks that the functions return an error.

There's a test for leading equals sign (below) that also checks the environment variable values.

Reply:
> The name of an environment variable cannot include an equal sign

It can as the first character. CMD.exe uses them as "hidden" environment variables. This is technically an implementation detail but we have to allow it otherwise, for example, copying environment variables from one environment to another would break.

Reply:
> There's a test for leading equals sign (below) that also checks the environment variable values.

Ok, looks like i missed that.


> This is technically an implementation detail but we have to allow it otherwise.

Should this be documented for Command's envs?

Reply:
I'm not sure we can make any promises here. As you saw, it's not documented by Microsoft. We can't really make strong guarantees than the official docs.

Reply:
No guarantees, just note about implementation detail.

Reply:
🤷I mean, maybe with appropriate caveats. I don't feel too strongly about it either way. I'm not sure what users can do with that information.

Reply:
https://github.com/python/cpython/blame/8e8a4baf652f6e1cee7acde9d78c4b6154539748/Modules/posixmodule.c#L12065-L12066 Oh, fun.